app/controllers/clark_kent/reports_controller.rb

```diff
@@ -96,4 +103,31 @@ def report_params
     params.require(:report).permit(:name, :resource_type, :sharing_scope_id, :sharing_scope_type)
   end
 
+  def parse_date(date_input)
+    begin
+      return american_parse(date_input)
+    rescue ArgumentError
+      begin
+        return Date.parse(date_input)
+      rescue ArgumentError
+        return nil
+      end
+    end
+  end
+
+  def american_parse(string)
+    string.gsub!("%2F", "/")
+    month,day,year = string.split('/').map{|part| part.to_i}
+    raise ArgumentError, "#{string} is not a valid date format" unless [month,day,year].all? { |c| !!c }
```

Comment:
Just a thought @edraut, if you raised anything other than ArgumentError here, you wouldn't need the nested begin / rescue blocks up in ````parse_date````